src/parser/transform/expression/transform_multi_assign_reference.cpp

```diff
@@ -0,0 +1,42 @@
+#include "duckdb/parser/parsed_expression.hpp"
+#include "duckdb/parser/transformer.hpp"
+
+namespace duckdb {
+
+unique_ptr<ParsedExpression> Transformer::TransformMultiAssignRef(duckdb_libpgquery::PGMultiAssignRef &root) {
+	// Multi assignment for the ROW function
+	if (root.source->type == duckdb_libpgquery::T_PGFuncCall) {
+		auto func = PGCast<duckdb_libpgquery::PGFuncCall>(*root.source);
+
+		// Explicitly only allow ROW function
+		char const *function_name =
+		    PGPointerCast<duckdb_libpgquery::PGValue>(func.funcname->tail->data.ptr_value)->val.str;
+		if (function_name == nullptr || strlen(function_name) != 3 || strncmp(function_name, "row", 3) != 0) {
+			return TransformExpression(root.source);
+		}
+
+		// Too many columns (ie. (x, y) = (1, 2, 3) )
+		if (root.ncolumns < func.args->length) {
+			throw ParserException("Too many columns for Multiple Assignment");
```

Comment:
Can this be more insightful as well, maybe something like:
"Could not perform multiple assignment, expected only %d values, found %d instead"